src/dotnet/commands/RestoringCommand.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.DotNet.Tools.MSBuild;
+using Microsoft.DotNet.Tools.Restore;
+
+namespace Microsoft.DotNet.Tools
+{
+    public class RestoringCommand : MSBuildForwardingApp
+    {
+        private bool NoRestore { get; }
+
+        private IEnumerable<string> ArgsToForward { get; }
+
+        private IEnumerable<string> ArgsToForwardToRestore
+        {
+            get
```

Comment:
This one is actually a less bad example because the linq is lazy. The ones that make a full array on property access are much worse. I still think this sort of thing should be a method because there's still too much redundant work (allocating objects to set up the lazy enumeration) if you call it more than once unnecessarily.